metagraph/src/graph/alignment/aligner_helper.cpp

```diff
@@ -382,6 +382,460 @@ void Alignment<NodeType>::append(Alignment&& other) {
+
+    switch (*cigar_it_) {
+        case Cigar::Operator::MATCH:
+        case Cigar::Operator::MISMATCH: {
+            assert(end_it_ > begin_it_);
+            assert(ref_end_it_ > ref_begin_it_);
+            assert(node_it_ > alignment_->begin());
+            score_ -= config_->get_row(*(end_it_ - 1))[*(ref_end_it_ - 1)];
+            --end_it_;
+            --ref_end_it_;
+
+            // if we're at the first node of the alignment, then traverse
+            // backwards to reconstruct a path
+            if (node_it_ - 1 == alignment_->begin()) {
+                auto last_node = prefix_node_ ? prefix_node_ : alignment_->front();
+                assert(last_node);
+                prefix_node_ = DeBruijnGraph::npos;
+                graph_->adjacent_incoming_nodes(last_node, [&](auto prev) {
+                    // TODO: what if there are multiple?
+                    if (!prefix_node_)
+                        prefix_node_ = prev;
+                });
+                ++offset_;
+            } else {
+                --node_it_;
+            }
+        } break;
+        case Cigar::Operator::DELETION: {
+            assert(end_it_ > begin_it_);
+            if ((--(cigar_it_.base())).offset()) {
+                score_ -= config_->gap_extension_penalty;
+            } else {
+                score_ -= config_->gap_opening_penalty;
+            }
+
+            --end_it_;
+        } break;
+        case Cigar::Operator::INSERTION: {
+            assert((--(cigar_it_.base())).get_it() >= alignment_->get_cigar().begin());
+            assert(ref_end_it_ > ref_begin_it_);
+            assert(node_it_ > alignment_->begin());
+            if ((--(cigar_it_.base())).offset()) {
+                score_ -= config_->gap_extension_penalty;
+            } else {
+                score_ -= config_->gap_opening_penalty;
+            }
+
+            --ref_end_it_;
+
+            // if we're at the first node of the alignment, then traverse
+            // backwards to reconstruct a path
+            if (node_it_ - 1 == alignment_->begin()) {
+                auto last_node = prefix_node_ ? prefix_node_ : alignment_->front();
+                assert(last_node);
+                prefix_node_ = DeBruijnGraph::npos;
+                graph_->adjacent_incoming_nodes(last_node, [&](auto prev) {
+                    // TODO: what if there are multiple?
+                    if (!prefix_node_)
+                        prefix_node_ = prev;
+                });
+                ++offset_;
+
+            } else {
+                --node_it_;
+            }
+        } break;
+        case Cigar::Operator::CLIPPED: { assert(false); }
+    }
+
+    ++cigar_it_;
+    ++trim_;
+
+    assert(cigar_rbegin_ <= cigar_it_);
+    assert(cigar_it_ <= cigar_rend_);
+
+    return *this;
+}
+
+template <typename NodeType>
+AlignmentPrefix<NodeType>& AlignmentPrefix<NodeType>::operator--() {
+    throw std::runtime_error("not implemented");
+    return *this;
```

Comment:
If this can't be implemented, it's better to remove this method.

Or, if this can be supported in the future, write "not implemented yet"